gems/aws-sdk-core/lib/aws-sdk-core/errors.rb

```diff
@@ -208,12 +208,40 @@ class InvalidProcessCredentialsPayload < RuntimeError; end
     # Raised when a client is constructed and region is not specified.
     class MissingRegionError < ArgumentError
       def initialize(*args)
-        msg = 'missing region; use :region option or '\
-              "export region name to ENV['AWS_REGION']"
+        msg = 'No region was provided. Configure the `:region` option or '\
+          "export the region name to ENV['AWS_REGION']"
         super(msg)
       end
     end
 
+    # Raised when a client is contsructed and the region is not valid.
+    class InvalidRegionError < ArgumentError
+      def initialize(*args)
+        super(<<-MSG)
```

Comment:
This is a fairly long exception message.  Do we have similar length messages anywhere else?  I do think it may be useful to customers though.  Maybe add it to the MissingRegionError as well?

Reply:
Mostly taken from here: https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-core/lib/aws-sdk-core/errors.rb#L226